src/release/bin/startup.sh

```diff
@@ -27,56 +31,59 @@ else
   exit 1
 fi
 
-if [ -z $GEOSERVER_HOME ]; then
+if [ -z "${GEOSERVER_HOME:-}" ]; then
   #If GEOSERVER_HOME not set then guess a few locations before giving
   # up and demanding user set it.
   if [ -r start.jar ]; then
      echo "GEOSERVER_HOME environment variable not found, using current "
      echo "directory.  If not set then running this script from other "
      echo "directories will not work in the future."
-     export GEOSERVER_HOME=`pwd`
+     GEOSERVER_HOME=$(pwd)
   else 
     if [ -r ../start.jar ]; then
       echo "GEOSERVER_HOME environment variable not found, using current "
       echo "location.  If not set then running this script from other "
       echo "directories will not work in the future."
-      export GEOSERVER_HOME=`pwd`/..
+      GEOSERVER_HOME="$(pwd)/.."
     fi
   fi 
 
-  if [ -z "$GEOSERVER_HOME" ]; then
+  if [ -z "${GEOSERVER_HOME:-}" ]; then
     echo "The GEOSERVER_HOME environment variable is not defined"
     echo "This environment variable is needed to run this program"
     echo "Please set it to the directory where geoserver was installed"
     exit 1
   fi
 
+  export GEOSERVER_HOME
 fi
 
-if [ ! -r "$GEOSERVER_HOME"/bin/startup.sh ]; then
+if [ ! -r "${GEOSERVER_HOME}/bin/startup.sh" ]; then
   echo "The GEOSERVER_HOME environment variable is not defined correctly"
   echo "This environment variable is needed to run this program"
   exit 1
 fi
 
 #Find the configuration directory: GEOSERVER_DATA_DIR
-if [ -z $GEOSERVER_DATA_DIR ]; then
-    if [ -r "$GEOSERVER_HOME"/data_dir ]; then
-        export GEOSERVER_DATA_DIR="$GEOSERVER_HOME"/data_dir
+if [ -z "${GEOSERVER_DATA_DIR:-}" ]; then
+    if [ -r "${GEOSERVER_HOME}/data_dir" ]; then
+        export GEOSERVER_DATA_DIR="${GEOSERVER_HOME}/data_dir"
     else
         echo "No GEOSERVER_DATA_DIR found, using application defaults"
 	      GEOSERVER_DATA_DIR=""
     fi
 fi
 
-cd "$GEOSERVER_HOME"
+cd "${GEOSERVER_HOME}" || exit
```

Comment:
if meant to signal a failure may be better to explicitly pass a >0 argument
```suggestion
cd "${GEOSERVER_HOME}" || exit 1
```

Reply:
Done